### PR TITLE
CityIO listener using interval, basemap init with cityIO data

### PR DESCRIPTION
**Pull request by RELNO**
cityIO has now a service and a component to control interfacing and avoid polluting `basemap` with cityio parsing

### CODE REVIEW

**Comment by yasushisakai** [comment]
looks good, (including the structure) so what should I do ? 😆  

one minor point:

- are you having a global tslint config? This might cause unnecessary file changes for the future. We want to use the same rule set. I see quote changes for the next PR.

**Comment by RELNO** [comment]
> looks good, (including the structure) so what should I do ? 😆

A lot :)


> * are you having a global tslint config? This might cause unnecessary file changes for the future. We want to use the same rule set. I see quote changes for the next PR.

Yes, let's do that, linting might create tons of changes upon save 

> * as a basic rule, do we want to squash the commits for each PR? reviewers point of view it is easier to track back, but from the request it may not be the case. (alternative: generally have fewer commits?)

Fewer commits is the way to go, I'll make sure to have smaller PRs from now on

File: src/app/cityio/cityio.component.ts [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+  `
+})
+export class CityioComponent implements OnInit {
+  constructor(private cityIOservice: CityioService) {}
+  title = "cityIO status...";
+  ngOnInit() {
+    this.getCityIOatInterval();
+  }
+
+  getCityIOatInterval() {
```
Comment (yasushisakai) [inline src/app/cityio/cityio.component.ts]: why should the component handle the interval? or maybe, what is this component’s roll?

File: src/app/cityio/cityio.service.ts [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+import { catchError, tap, retry } from "rxjs/operators";
+
+const CITYIO_TABLE_URL = "https://cityio.media.mit.edu/api/table/";
+const DEFAULT_TABLE_NAME = "grasbrook";
+
+@Injectable({
+  providedIn: "root"
+})
+export class CityioService {
+  cityIOTableURL: string;
```
Comment (yasushisakai) [inline src/app/cityio/cityio.service.ts]: type URL?

Comment (RELNO) [reply src/app/cityio/cityio.service.ts]: how can `"./assets/cityIO_demo.json"` be returned as Url type?

**Pull request merged**